toolkit/tools/pkg/imagecustomizerlib/packageinformation.go

```diff
@@ -0,0 +1,204 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+)
+
+type PackageVersionInformation struct {
+	PackageVersionComponents []uint64 `yaml:"PackageVersionComponents"`
+	PackageRelease           uint32   `yaml:"PackageRelease"`
+	DistroName               string   `yaml:"DistroName"`
+	DistroVersion            uint32   `yaml:"DistroVersion"`
+}
+
+func (pi *PackageVersionInformation) getVersionString() string {
+	var version string
```

Comment:
Ideally you should use [strings.Builder](https://pkg.go.dev/strings#Builder) here.

Reply:
pending

Reply:
fixed.